plugins/defaultplugins/srplugin/vppcalls/srv6.go

```diff
@@ -0,0 +1,510 @@
+// Copyright (c) 2018 Bell Canada, Pantheon Technologies and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vppcalls
+
+import (
+	"fmt"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/ligato/cn-infra/logging"
+	"github.com/ligato/cn-infra/logging/measure"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/common/bin_api/sr"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/common/model/srv6"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/ifplugin/ifaceidx"
+)
+
+// Constants for behavior function hardcoded into VPP (there can be also custom behavior functions implemented as VPP plugins)
+// Constants are taken from VPP's vnet/srv6/sr.h (names are modified to Golang from original C form in VPP code)
+const (
+	BehaviorEnd    uint8 = iota + 1 //Behavior of simple endpoint
+	BehaviorX                       //Behavior of endpoint with Layer-3 cross-connect
+	BehaviorT                       //Behavior of endpoint with specific IPv6 table lookup
+	BehaviorDfirst                  //Unused. Separator in between regular and D
+	BehaviorDX2                     //Behavior of endpoint with decapulation and Layer-2 cross-connect (or DX2 with egress VLAN rewrite when VLAN notzero - not supported this variant yet)
+	BehaviorDX6                     //Behavior of endpoint with decapsulation and IPv6 cross-connect
+	BehaviorDX4                     //Behavior of endpoint with decapsulation and IPv4 cross-connect
+	BehaviorDT6                     //Behavior of endpoint with decapsulation and specific IPv6 table lookup
+	BehaviorDT4                     //Behavior of endpoint with decapsulation and specific IPv4 table lookup
+	BehaviorLast                    //seems unused, note in VPP: "Must always be the last one"
+)
+
+// Constants for steering type
+// Constants are taken from VPP's vnet/srv6/sr.h (names are modified to Golang from original C form in VPP code)
+const (
+	SteerTypeL2 uint8 = iota*2 + 2
```

Comment:
This might be little harder to figure out. I would suggest to set each `const` to specific value so it can be read more easily.

Reply:
fixed in 063ec321491b198a444a3061db96e3a1337685b1